docs/proposals/PROPOSAL_E.md

```diff
@@ -0,0 +1,296 @@
+  "mediaType": "application/vnd.oci.image.manifest.v1+json",
+  "config": { ... },
+  "layers": [ ... ],
+  "reference": { // new field
+    "mediaType": "application/vnd.oci.image.manifest.v1+json", // any manifest media type
+    "size": 1234,
+    "digest": "sha256:a1a1a1...",
+    "annotations": []
+  },
+  "annotations": [
+    // annotations for this artifact here
+  ]
+}
+```
+
+#### Artifact Spec
+
+Create a new artifact media type to support future use cases where a separate config blob, and an ordered list of blobs, may not match an artifact's use case:
+
+```jsonc
+{
+  "schemaVersion": 2,
+  "mediaType": "application/vnd.oci.artifact.manifest.v1+json",
+  "blobs": [
+    // optional list, ordering is not enforced by the spec
+    // but may be required for specific artifact types
+  ],
+  "reference": { // optional
+    // include reference here
+  },
+  "annotations": [ // optional
+    // annotations for this artifact here
+  ]
+}
+```
+
+### Registry HTTP API
+
+#### References API
+
+Add the following `references` API (`<ref>` is the digest in the `reference` schema field):
+
+```text
+GET /v2/<name>/references/<ref>
+```
+
+The response is an Index of descriptors:
+
+```jsonc
+{
+  "schemaVersion": 2,
+  "mediaType": "application/vnd.oci.image.index.v1+json",
+  "manifests": [
+    {
+      "mediaType": "application/vnd.oci.image.manifest.v1+json",
+      "size": 1234,
+      "digest": "sha256:a1a1a1...",
+      "annotations": [
+        // annotations pulled up from manifest
+        "org.opencontainers.artifact.type": "example/icecream",
+        "org.opencontainers.artifact.created": "2022-01-01T14:42:55Z",
+        "org.example.icecream.flavor": "chocolate"
+      ]
+    },
+    {
+      "mediaType": "application/vnd.oci.image.manifest.v1+json",
+      "size": 1234,
+      "digest": "sha256:a2a2a2...",
+      "annotations": [
+        "org.opencontainers.artifact.type": "example/icecream",
+        "org.opencontainers.artifact.created": "2022-01-01T15:24:30Z",
+        "org.example.icecream.flavor": "vanilla"
+      ]
+    }
+  ],
+  "annotations": [
+    "org.opencontainers.references.params": "..." // optional base64 encoded parameters
+  ]
+}
+```
+
+If a registry does not implement the `references` API, it MUST return a 404.
+If a query results in no references found, an empty manifest list MUST be returned.
+
+#### Ordering and Pagination
+
+- The registry MUST order the results in a way that allows pagination.
+- Adding the `n` query parameter is used to limit the number of entries per index returned.
+- The registry SHOULD return fewer than `n` results when the generated Index would exceed the recommended maximum manifest size.
+- The `Link` HTTP header is included in the response when additional results are available and is set to the URL for the next page of results.
+
+#### Filtering and Sorting
+
+- The registry SHOULD support filtering by annotation using the `filter=<field><op><value>` parameter.
```

Comment:
Now that we have a working group for filtering, should we be defining details here?

Reply:
I think we've only proposed it internally, still a long way from having another OCI WG repo, but I'm happy to track these ideas where ever it makes the most sense. I think we can break up the changes recommended by this proposal to near term and longer term. So if we want to push through image-spec changes and define digest tags formally, those should be possible while we still work out how to define this API and come up with a filtering solution.